### ISSUE TITLE
Completion list for module with default export should use "default", not the function's name

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#12957]
**TypeScript Version:**  nightly (2.2.0-dev.20161215)

**Code**

```ts
/// <reference path="fourslash.ts" />

// @Filename: a.ts
////export function x() {}
////export default function y() {}

// @Filename: b.ts
////import * as a from "./a";
////a./**/;

goTo.marker();
verify.completionListContains("x", "function x(): void");
verify.completionListContains("default", "function y(): void");
```

**Expected behavior:**

Test succeeds.

**Actual behavior:**

```
Error: Marker: 
Expected "{
  "name": "default",
  "text": "function b(): void"
}" to be in list [{
  "name": "x",
  "kind": "function"
},
{
  "name": "y",
  "kind": "function"
}]
```

Problem: `a.y()` won't compile, `a.default()` will. So we should use `default` for the completion, not `y`.

**Comment by DanielRosenwasser** [comment]
Looks like a regression of #6124.

**Comment by andy-ms** [comment]
Fixed by #16742.
